connect-inject/health_check_resource_test.go

```diff
@@ -0,0 +1,566 @@
+					Namespace: "default",
+					Labels:    map[string]string{labelInject: "true"},
+				},
+				Spec: testPodSpec,
+				Status: corev1.PodStatus{
+					HostIP: "127.0.0.1",
+					Phase:  corev1.PodRunning,
+					Conditions: []corev1.PodCondition{{
+						Type:   corev1.PodReady,
+						Status: corev1.ConditionTrue,
+					}},
+				},
+			},
+			nil,
+			"",
+		},
+		{
+			"PodRunning no Ready Status will be ignored for processing",
+			false,
+			"",
+			&corev1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      testPodName,
+					Namespace: "default",
+					Labels:    map[string]string{labelInject: "true"},
+				},
+				Spec: testPodSpec,
+				Status: corev1.PodStatus{
+					HostIP: "127.0.0.1",
+					Phase:  corev1.PodRunning,
+				},
+			},
+			nil,
+			"",
+		},
+	}
+	for _, tt := range cases {
+		t.Run(tt.Name, func(t *testing.T) {
+			var err error
+			require := require.New(t)
+			// Get a server, client, and handler
+			server, client, resource := testServerAgentResourceAndController(t, tt.Pod)
+			defer server.Stop()
+			// register the service with Consul
+			server.AddService(t, testServiceNameReg, api.HealthPassing, nil)
+			if tt.PreCreateHealthCheck {
+				// register the health check if this is not an object create path
+				registerHealthCheck(t, client, tt.InitialState)
+			}
+			// Upsert and Reconcile both use this common function to reconcile the pod
+			err = resource.reconcilePod(tt.Pod)
+			// if we're expecting any error from reconcilePod
+			if tt.Err != "" {
+				// used in the cases where we're expecting an error from
+				// the controller/handler, in which case do not check agent
+				// checks as they're not relevant/created.
+				require.Error(err, tt.Err)
+				return
+			}
+			require.NoError(err)
+			// get the agent checks if they were registered
+			actual := getConsulAgentChecks(t, client)
+			require.True(cmp.Equal(actual, tt.Expected, cmpopts.IgnoreFields(api.AgentCheck{}, ignoredFields...)))
+		})
+	}
+}
+
+func TestUpsert_PodWithNoServiceReturnsError(t *testing.T) {
+	t.Parallel()
+	require := require.New(t)
+	pod := &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      testPodName,
+			Namespace: "default",
+			Labels:    map[string]string{labelInject: "true"},
+			Annotations: map[string]string{
+				annotationStatus:  injected,
+				annotationService: testServiceNameAnnotation,
+			},
+		},
+		Spec: testPodSpec,
+		Status: corev1.PodStatus{
+			HostIP: "127.0.0.1",
+			Phase:  corev1.PodRunning,
+			Conditions: []corev1.PodCondition{{
+				Type:   corev1.PodReady,
+				Status: corev1.ConditionTrue,
+			}},
+		},
+	}
+	server, _, resource := testServerAgentResourceAndController(t, pod)
+	defer server.Stop()
+	// start Upsert, it will attempt to reconcile the Pod but the service doesnt exist in Consul so will fail
+	err := resource.Upsert("", pod)
+	require.EqualError(err, "unable to register health check: Unexpected response code: 500 (ServiceID \"test-pod-test-service\" does not exist)")
```

Comment:
This should fix the failing ent test. The issue is that in oss the error is as ☝️ , but if running against enterprise service ID is prefixed with a namespace. 
```suggestion
	require.Contains(err.Error(), "test-pod-test-service\" does not exist)")
```